### DISCUSSION TITLE
EJBCA version 7.4.3.2 still supported?

### DISCUSSION THREAD

**Discussion by ramiabdallah1512** [Keyfactor/ejbca-ce#546] [Q&A]
Hello Tomas,

Thank you for your feedback

However, I need some additional information: is version 7.4.3.2 still supported?

Upgrading to version 8 would present major risks for my system.

I would be grateful if you could confirm whether version 7.4.3.2 is still supported and whether I can continue to use it as a production version.

Best regards,

**Comment by primetomas** [comment]
What do you mean with supported exactly? This is a community forum. EJBCA Community has no official, guaranteed, support, it is supported by all the members of this forum on a best, not payed, effort. So if someone is able and willing to answer your questions you will get answer, otherwise not. It is. Very simple. I'll answer questions at best as I can, when I have time over (and the same for others).

If you need professional support with a SLA you can contact Keyfactor for an EJBCA Enterprise contract. You can contact Keyfactor to ask about which versions are commercially supported.
https://www.keyfactor.com/products/ejbca-enterprise/

Best regards,
Tomas

**Comment by svenska-primekey** [comment]
7.4.3.2 is over 2 years old now. EJBCA now runs with a minimum of Java 11 and Wildfly 26 is recommended. In my opinion upgrading is highly recommended.

**Comment by primetomas** [comment]
EJBCA 7.4.3.2 does have known CVEs.
https://www.cvedetails.com/version-list/23203/83348/1/Primekey-Ejbca.html

**Comment by ramiabdallah1512** [comment]
Thank you for your answers, 

Tomas.
I've just looked at the link you sent me. The version affected is the Enterprise edition, but my edition is Community. Can we consider the same thing?

To be honest, I've seen that upgrading to version 8 is going to cause a number of problems because it's very complex. Above all, the migration procedure is not clear.

**Reply by primetomas** [reply]
Yes the code base for Community is just a subset of Enterprise. You have to consider these.

Unless you have done custom implementation against internal APIs, the upgrade should not be very complex. It's fairly well documented here and upgrades can be very smooth if you have a process for it. Same as you need for OS, database, Java, etc, upgrades.
https://doc.primekey.com/ejbca/ejbca-installation/upgrading-ejbca

**Comment by rabdallah1512** [comment]
Thank you for your quick response. I have decided to upgrade EJBCA Community from version 7.4.3.2 to version 8.2.0.1. I have a question regarding the upgrade:
My current version of WildFly is 10.1.0. Is it compatible with EJBCA 8.2.0.1?
If so, can I simply copy the conf/*.properties and p12 files to the appropriate directory, then run the ant deployear command?
Do I also need to run the ant deploy-keystore command?
I don't want to upgrade WildFly if the current version is compatible.

**Reply by svenska-primekey** [reply]
No, you will need to upgrade Wildfly to 26.1.3 to use EJBCA 8.2.0.1. Once you have wildfly updated you can use the ant deployear command. You can re-use the keystore files with the upgraded wildfly. You would only need to run ant deploy-keystore if you did not move the keystore files to the upgraded wildfly configuration directory.

You should upgrade wildflly, version 10 is super old.

**Comment by ramiabdallah1512** [comment]
Thank you for your reply.
Question: which keystore do I need to move and under which directory do I
need to move them?

**Reply by svenska-primekey** [reply]
wildfly/standalone/configuration there should be a directory called keystore in there. Copy that keystore directory to updated wildfly configuration directory.
